sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -66,6 +66,10 @@ int max98373_trigger(struct snd_pcm_substream *substream, int cmd)
 	int j;
 	int ret = 0;
 
+	/* set spk pin by playback only */
+	if (substream->stream == SNDRV_PCM_STREAM_CAPTURE)
```

Comment:
@RanderWang this change will work. But I have an alternate suggestion. I think the reason snd_soc_dai_active() doesnt work is because it is not decremented until the the pcm is in closed in snd_pcm_close() gets invoked. So instead of enabling/disabling the pin in the trigger op, we should do this in the startup/shutdown ops for the dai link. what do you think? @RDharageswari FYI

Reply:
@ranj063 doesn't this impact suspend usages? You'd want to disable the pin prior to suspend, no?

Reply:
@plbossart thats right. This will impact the suspend case. I get the feeling that something might be missing in the DAPM power sequence but i dont have the time to dig into it right now. So this change is good for now.

Reply:
so do we want to merge to the SOF tree but keep it on the back burner in case we find a better fix?

Reply:
@plbossart yes, I think that would be a good idea. I'll find some time to investigate more in the next 2-3 weeks and then let you know if I can find a better fix.

Reply:
@ranj063  I share some info about this codec. It is tricky that 98373 define three signal generators https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/codecs/max98373.c#L78-L80, which are endpoints defined by ASoC. And these endpoints are connected to speaker, another endpoint, therefore codec is always active. @RDharageswari disable spk pin to break the path at reasonable point so that codec can get suspended.